code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -784,42 +798,94 @@
 	drop_organs()
 	return ..()
 
-/**
- * Calculates how much blood this limb is losing per life tick
- *
- * Arguments:
- * * ignore_modifiers - If TRUE, ignore the bleed reduction for laying down and grabbing your limb
- */
-/obj/item/bodypart/proc/get_part_bleed_rate(ignore_modifiers = FALSE)
+/// INTERNAL PROC, DO NOT USE
+/// Properly sets us up to manage an inserted embeded object
+/obj/item/bodypart/proc/_embed_object(obj/item/embed)
+	if(embed in embedded_objects) // go away
+		return
+	// We don't need to do anything with projectile embedding, because it will never reach this point
+	RegisterSignal(embed, COMSIG_ITEM_EMBEDDING_UPDATE, .proc/embedded_object_changed)
+	embedded_objects += embed
+	refresh_bleed_rate()
+
+/// INTERNAL PROC, DO NOT USE
+/// Cleans up any attachment we have to the embedded object, removes it from our list
+/obj/item/bodypart/proc/_unembed_object(obj/item/unembed)
+	UnregisterSignal(unembed, COMSIG_ITEM_EMBEDDING_UPDATE)
+	embedded_objects -= unembed
+	refresh_bleed_rate()
+
+/obj/item/bodypart/proc/embedded_object_changed(obj/item/embedded_source)
+	SIGNAL_HANDLER
+	/// Embedded objects effect bleed rate, gotta refresh lads
+	refresh_bleed_rate()
+
+/// Sets our generic bleedstacks
+/obj/item/bodypart/proc/setBleedStacks(set_to)
+	adjustBleedStacks(set_to - generic_bleedstacks)
```

Comment:
```suggestion
/obj/item/bodypart/proc/setBleedStacks(set_to)
	SHOULD_CALL_PARENT(TRUE)
	adjustBleedStacks(set_to - generic_bleedstacks)
```

Reply:
Why? I'm perfectly fine with someone overriding this if they really wanted to. Is it because they should instead be overriding adjustBleedStacks? Is that worth the extra line?

Reply:
Yes.

Reply:
I'm not sure I agree, but alright

Reply:
actually I just like this, good point